lib/ansible/plugins/process/forking.py

```diff
@@ -40,25 +53,117 @@
 from ansible.executor.task_executor import TaskExecutor
 from ansible.executor.task_result import TaskResult
 from ansible.module_utils._text import to_text
+from ansible.module_utils.six.moves import queue as Queue
+from ansible.plugins.process import ProcessModelBase, ResultsSentinel
 from ansible.utils.display import Display
 
-__all__ = ['WorkerProcess']
 
 display = Display()
 
 
+def results_thread_main(pm):
+    while True:
+        try:
+            result = pm._worker_results_q.get()
+            if isinstance(result, ResultsSentinel):
+                break
+            else:
+                pm._final_q_lock.acquire()
+                pm._final_q.append(result)
+                pm._final_q_lock.release()
+        except (IOError, EOFError) as e:
+            break
+        except Queue.Empty:
+            pass
+
+
+__all__ = ['ProcessModel',]
+
+
+class ProcessModel(ProcessModelBase):
```

Comment:
Not sure if it would be useful for something later, but should we have a class attr with the name of the plugin in it?  Some plugin types do this others don't.

Reply:
_load_name gets appended by loader itself